components/bl-pdfViewer-component/src/subcomponents.js

```diff
@@ -57,62 +67,93 @@ export function Controls(props) {
     }
   };
 
-  useEffect(() => {
-    setNextPage(currentPage);
-  }, [currentPage]);
-
-  if (!pageCount) {
-    return null;
-  }
+  const onNextButtonClick = () => setCurrentPage(state => state + 1);
+  const onPrevButtonClick = () => setCurrentPage(state => state - 1);
 
   return (
-    <div ref={ controlsRef } className="controls">
+    <div className="controls__page">
       <button
-        className="controls-button"
-        onClick={ () => setCurrentPage(state => state - 1) }
+        className="controls__button"
+        onClick={ onPrevButtonClick }
         disabled={ (currentPage > pageCount) || (currentPage <= 1) }>
         <PrevButtonIcon/>
       </button>
-      <div className="pages-info">
-        <input
-          className="page-input"
-          value={ nextPage }
-          onChange={ onInputChange }
-          onKeyDown={ onEnter }
-          onBlur={ onSubmit }
-        />
-        <span>/{ pageCount }</span>
-      </div>
       <button
-        className="controls-button"
-        onClick={ () => setCurrentPage(state => state + 1) }
+        className="controls__button"
+        onClick={ onNextButtonClick }
         disabled={ (currentPage >= pageCount) || (currentPage <= 0) }>
         <NextButtonIcon/>
       </button>
+      <input
+        className="controls__page-input"
+        value={ nextPage }
+        onChange={ onInputChange }
+        onKeyDown={ onEnter }
+        onBlur={ onSubmit }
+      />
+      <span>/{ pageCount }</span>
     </div>
   );
 }
 
-function NextButtonIcon() {
+function ScaleControls({ scale, setScale }) {
+  const onMinusScaleClick = () => setScale(s => +Math.max(0.1, s - 0.1).toFixed(1));
+  const onPlusScaleClick = () => setScale(s => +Math.min(10, s + 0.1).toFixed(1));
+
   return (
-    <svg
-      className="controls-button-icon"
-      focusable="false"
-      aria-hidden="true"
-      viewBox="0 0 24 24">
-      <path d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"></path>
-    </svg>
+    <div className="controls__scale">
+      <button className="controls__button" onClick={ onMinusScaleClick } disabled={ scale <= 0.1 }>
+        <MinusIcon/>
+      </button>
+      <span>{ scale }</span>
+      <button className="controls__button" onClick={ onPlusScaleClick } disabled={ scale >= 10 }>
+        <PlusIcon/>
+      </button>
+    </div>
+  );
+}
+
+function OtherControls({ pdfUrl, showDownloadButton, showPrintButton }) {
+  const printButtonClick = () => printJS({
+    printable         : pdfUrl,
+    type              : 'pdf',
+    onPrintDialogClose: () => document.getElementById('printJS').remove(),
```

Comment:
Is it works correct if we have multiple pdf files on the screen?

Reply:
Yes. This element is mounted when the print dialog is opened, and at that moment there is no way to open this window for another component. So when this event fires, there can only be one element